server/model/monitor.js

```diff
@@ -613,6 +614,45 @@ class Monitor extends BeanModel {
 
                 } else if (this.type === "port") {
                     bean.ping = await tcping(this.hostname, this.port);
+                    if (this.isEnabledExpiryNotification()) {
+                        const host = this.hostname;
+                        const port = this.port || 443;
+                        try {
+                            const options = {
+                                host,
+                                port,
+                                servername: host,
+                            };
+
+                            // Convert TLS connect to a Promise and await it
+                            const tlsInfoObject = await new Promise((resolve, reject) => {
+                                const socket = tls.connect(options);
+
+                                socket.on("secureConnect", () => {
+                                    try {
```

Comment:
Please have a look at how snmp handles the socket-cleanup.

https://github.com/louislam/uptime-kuma/blob/2e2acbd48af4aec981a7d47c9d52b1e80d26f86d/server/monitor-types/snmp.js